### DISCUSSION TITLE
Add Appium UI tests to Repo

### DISCUSSION THREAD

**Discussion by ne0rrmatrix** [CommunityToolkit/Maui#2122] [New Feature Discussions]
I have been thinking we need a good way to test UI and device specific issues with all of code in our project. I have a link here where I have added Appium UI tets for media element to test all the buttons and UI interactions that we have for each device. I am hoping we can add this to toolkit for both internal use and the show the community how to setup and deploy tests for dotnet maui apps. TY to Gerald for the MS blog providing the samples I needed to get this working :) 

Here is sample of changes. https://github.com/CommunityToolkit/Maui/compare/main...ne0rrmatrix:MauiOld:AddUITests

**Comment by Pastajello** [comment]
@TheCodeTraveler 
Has there been any motion for this?

With all the recent changes it'd be nice to see whether something broke.

I've heard Appium tests can be run using:

https://github.com/jonathanpeppers/cuttlefish

with this common scenarios for (just my recent interest) for example changing status bar colors could be easily verified on different APIs with screen shot tests.

**Reply by VladislavAntonyuk** [reply]
I would suggest Maestro, as it is much easier to create and run
